reactor-netty-http/src/main/java/reactor/netty/http/server/HttpServerRoutes.java

```diff
@@ -262,6 +264,15 @@ default HttpServerRoutes put(String path,
 	HttpServerRoutes route(Predicate<? super HttpServerRequest> condition,
 			BiFunction<? super HttpServerRequest, ? super HttpServerResponse, ? extends Publisher<Void>> handler);
 
+	/**
+	 * A comparator used to sort HttpRouteHandlers.If this method called more times,
```

Comment:
also mention that currently it sorts the existing routes in place upon calling this method, then sorts them again each time a new route is added.

Reply:
we should also think about how on can go back to the default if a `Comparator` has previously been set. the most straightforward way is to make the `Comparator` nullable and pass `null` to remove any custom comparator. wdyt @violetagg ?

(A null-free way would be to add "insertion index" to the routeHandler's metadata, so that one can provide a `Comparator<HttpRouteHandlerMetadata>` restoring insertion order, but that's adding a field for a corner case.)

Reply:
we can either allow `null` in the API so that this will restore the default behaviour or add API similar to what we have for the security `HttpServer#secure(...)` vs. `HttpServer#noSSL`

Reply:
I personally like the second approach if it works for you (avoiding `null`-based approaches where possible)

Reply:
add a new method `noCompare()`?

Reply:
May be `noComparator()` or `disableComparator()`

Reply:
change to `noComparator()` like `HttpServer#noSSL`